src/Pages/MyCart/MyCart.js

```diff
@@ -0,0 +1,160 @@
+import react, { Component } from "react";
+import SelleProduct from "./SelleProduct";
+import configData from "../../config.json";
+import "./MyCart.scss";
+
+class MyCart extends Component {
+  constructor() {
+    super();
+    this.state = { allChekced: true };
+  }
+  componentDidMount() {
+    fetch(configData.MYCART, {
+      headers: {
+        Authorization: localStorage.getItem("key"),
+      },
+    })
+      .then((res) => res.json())
+      .then((res) => this.setState({ myCartData: res.data }));
+  }
+
+  changeQuantity = (e, product) => {
+    const { value, dataset } = e.target;
+    const myCartData = [];
+    this.state.myCartData.forEach((el) => {
+      if (el.orderlist_id === product.orderlist_id) {
+        if (dataset.state === "+") {
+          el.quantity = parseInt(el.quantity) + 1;
+        } else if (dataset.state === "-") {
+          el.quantity = parseInt(el.quantity) - 1;
+        } else {
+          el.quantity = value;
+        }
+      }
+
+      myCartData.push(el);
+    });
+    this.setState({ myCartData });
+  };
+
+  checkProduct = (e, product) => {
+    const { checked } = e.target;
+    const { orderlist_id, chekced } = product;
+
+    const myCartData = [];
+    let checkAllCheck = true;
+    this.state.myCartData.forEach((el) => {
+      el.orderlist_id === product.orderlist_id && (el.checked = !el.checked);
+      el.checked !== checkAllCheck && (checkAllCheck = false);
+      myCartData.push(el);
+    });
+    this.setState({ myCartData, allChekced: checkAllCheck });
+  };
```

Comment:
이 로직으로 기능은 구현을 잘 할 수 있으나 불필요하게 for문이 배열의 처음부터 끝까지 도는 것 같습니다.
예를 들어서 제일 마지막에 있는 제품을 체크한다고 가정했을 때 그 이전의 모든 제품들을 체크하면서 와야 합니다.
그리고 기존 데이터를 재활용해서 사용하는 방법이 좋을 것 같습니다.
또한 모든 체크박스가 체크 되어 있는지 확인하는 로직이 한 번에 이해하기 어렵게 짜셔서 다음과 같이 하는 방법도 확인하시면 좋을 것 같습니다.

```suggestion
 checkProduct = (e, product) => {
    const { myCartData } = this.state;
    const copiedData = [...myCartData];
    for (let i in copiedData) {
      if (copiedData[i].orderList_id === product.orderList_id) {
        copiedData[i].checked = !copiedData[i].checked;
        break;
      }
    }
    const allChekced = copiedData.every(el => el.checked);
    this.setState({ myCartData: copiedData, allChekced });
  };
```

Reply:
해보니 제대로 작동이 되지 않아 코드를 조금 효율적으로 줄이는 방법은 알았지만 제대로 작동이 되지 않내요 ㅜㅜ